src/components/batch_select.vue

```diff
@@ -0,0 +1,174 @@
+             size="large"
+             click_outside_to_close>
+        <div class="modal-header">
+          Select Items ({{ d_selected_items.length }} out of
+          {{ choices.length }} items selected)
+        </div>
+        <input class="input batch-search-field"
+               type="text"
+               placeholder="Enter a name"
+               v-model="d_batch_search_query"
+        />
+        <div>
+          <ul class="batch-select-card-grid">
+            <li class="batch-select-card"
+                :class="{
+                  selected: d_selected_items.some((el) => are_items_equal(el, item)),
+                }"
+                v-for="item of batch_filtered_items"
+                :key="item.pk"
+                @click="batch_toggle_select(item)">
+              <slot v-bind:item="item"></slot>
+            </li>
+          </ul>
+        </div>
+        <div class="button-footer-right modal-button-footer">
+          <button class="modal-confirm-button"
+                  @click="confirm_selection">
+            Confirm
+          </button>
+          <button class="modal-cancel-button"
+                  @click="d_show_batch_select_modal = false">
+            Cancel
+          </button>
+        </div>
+      </modal>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+
+import _ from 'lodash';
+
+import Modal from '@/components/modal.vue';
+
+// types for function props
+type Comparator = (lhs: unknown, rhs: unknown) => boolean;
+type Filter = (item: unknown, filter_text: string) => boolean;
+
+@Component({
+  components: {
+    Modal
+  }
+})
+export default class BatchSelect extends Vue {
+  @Prop({ required: true, type: Array })
+  choices!: unknown[];
+
+  @Prop({ type: Array })
+  value!: unknown[];
+
+  @Prop({ type: Function })
+  are_items_equal!: Comparator;
+
+  @Prop({ type: Function })
+  filter_fn!: Filter;
+
+  d_show_batch_select_modal = false;
+  d_batch_search_query: string = "";
+  d_selected_items: unknown[] = [];
+
+  created() {
+    this.d_selected_items = this.value.slice();
+  }
+
+  @Watch("value", { deep: true })
+  on_value_change(new_value: unknown[], old_value: unknown[]) {
+    this.d_selected_items = this.value.slice();
+  }
+
+  batch_toggle_select(item: unknown) {
+    if (_.some(this.d_selected_items, (el) => this.are_items_equal(el, item))) {
+      this.d_selected_items = _.filter(
+        this.d_selected_items,
+        (el) => !this.are_items_equal(el, item)
+      );
+    }
+    else {
+      this.d_selected_items.push(item);
+    }
+  }
+
+  get batch_filtered_items() {
+    return _.isEmpty(this.d_batch_search_query) ?
+      this.choices :
+      _.filter(
+        this.choices,
+        (item) => this.filter_fn(item, this.d_batch_search_query)
+      );
+  }
+
+  // Call when user clicks the "confirm" button.
+  confirm_selection() {
+    this.$emit("items_selected", this.d_selected_items);
```

Comment:
Sorry I didn't catch this earlier: this event should be called "input" so that the component properly supports v-model. See https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components

Then, the calls to `set_student_files` and `set_instructor_files` can be replaced with `$emit("field_change", d_suite)`. I'll mark those specific places just do be clear.

Reply:
Fixed